sql/updates/latest-dev.sql

```diff
@@ -1,2 +1,88 @@
 DROP FUNCTION _timescaledb_internal.time_col_name_for_chunk(name,name);
 DROP FUNCTION _timescaledb_internal.time_col_type_for_chunk(name,name);
+
+-- Handle column renames for continuous aggregates that were not
+-- handled correctly and fix it in the update. We save the information
+-- in a table.
+CREATE TABLE rename_tables (
```

Comment:
Not sure what you mean. If you are referring to a CTE, then no, if you are referring to using the `TEMPORARY` keyword, it does not offer any advantaged: we still need to drop the table after we're done with it.

Reply:
IMHO it should be at least UNLOGGED: `CREATE UNLOGGED TABLE ...`

Reply:
@mkindahl The whole point with a temporary table is that it is automatically dropped. IMO, a TEMPORARY table offers several advantages. If it is specified with ON COMMIT DROP, the table is automatically dropped at the end of the transaction. But I think the most important part of making it TEMPORARY is that it clearly signals here that this is something that should not stick around, which makes it easier to understand what is expected. The first thing I had to do as a reviewer is to figure out what this table would do and whether it would stick around post update. If it had been marked TEMPORARY ... ON COMMIT DROP, it would have been clear that this is a table that is only for this script.

Also adding UNLOGGED is a good idea, as @fabriziomello points out.

Reply:
If I modify the update script as you suggest and try to do an update, here is the result:

```
mats=# alter extension timescaledb update to "2.5.0-dev";
ERROR:  cannot drop table rename_tables because extension timescaledb requires it
HINT:  You can drop extension timescaledb instead.
```

In other words, we need to drop the table anyway and there is no improvement of using `TEMPORARY`.

Reply:
The table is made part of the extension since you add it in the update script, so one would have to do `ALTER EXTENSION name DROP member_object` to disassociate it and it would work. One advantage of that, as opposed to an explicit DROP is that you can do it in the same context as where the table is created.

But you can decide yourself what you think it best. I do think the improvement is still clarity.

Reply:
I will stick to the tested and working approach, since it is significantly simpler.

Reply:
> IMHO it should be at least UNLOGGED: `CREATE UNLOGGED TABLE ...`

This is a comparably small table, so I do not see the advantage of it, updates are not really expected to be a performance bottleneck, but sure, it was added.